tests/test_train.py

```diff
@@ -134,20 +134,34 @@ def test_config_file(tmp_path: Path) -> None:
     subprocess.run(args, check=True)
 
 
-@pytest.mark.parametrize("task", ["cowc_counting", "cyclone", "sen12ms", "landcoverai"])
+@pytest.mark.parametrize(
+    "task",
+    [
+        "bigearthnet",
+        "byol",
+        "chesapeake_cvpr",
+        "cowc_counting",
+        "cyclone",
+        "landcoverai",
+        "naipchesapeake",
+        "resisc45",
+        "sen12ms",
+        "so2sat",
+        "ucmerced",
+    ],
+)
 def test_tasks(task: str, tmp_path: Path) -> None:
     output_dir = tmp_path / "output"
-    data_dir = os.path.join("tests", "data", task)
     log_dir = tmp_path / "logs"
     args = [
         sys.executable,
         "train.py",
         "experiment.name=test",
         "program.output_dir=" + str(output_dir),
-        "program.data_dir=" + data_dir,
         "program.log_dir=" + str(log_dir),
         "trainer.fast_dev_run=1",
         "experiment.task=" + task,
         "program.overwrite=True",
+        "config_file=" + os.path.join("conf", "task_defaults", task + ".yaml"),
```

Comment:
@calebrob6 without this line, the default configs aren't being loaded, so there might be something broken with default loading.

Reply:
The default config is loaded by name here -- https://github.com/microsoft/torchgeo/blob/main/train.py#L97.

Reply:
Yes, but if you comment out this line, you'll see that it isn't loading the default config.

Reply:
It definitely is loading the default config, else it would throw an error about not finding the default config

Reply:
Because of the way configurations are merged, `experiment.datamodule.root_dir` was being overridden by the defaults in `defaults.conf`

Reply:
Yeah, I realize I've been abusing the config system by modifying it to suit the tests. What if we:

1. Keep doing this until we get the release out since the `conf` directory won't get installed anyway
2. After the release is out, create a `tests/conf` directory for test-specific configuration settings
3. Think about #227 post-rebuttal when we have more time

We could also do 2 real quick, it wouldn't take that long.

Reply:
Can you make a TODO here or somewhere that we'll remember?

Reply:
I'll create an issue so we don't re-trigger the tests